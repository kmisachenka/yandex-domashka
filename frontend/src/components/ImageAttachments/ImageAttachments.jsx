import React from 'react';
import * as PropTypes from 'prop-types';

import styles from './ImageAttachments.module.scss';
import { ReactComponent as ImageSVG } from './svg/image.svg';

export default function ImageAttachments(props) {
  const { attachments } = props;
  return (
    <>
      <ImageSVG className={styles.svg} />
      <ul className={styles.list}>
        {attachments.map(({ url }) => (
          <li key={url} className={styles.item}>
            <img src={`/img/${url}`} className={styles.image} alt="Картинка" />
          </li>
        ))}
      </ul>
    </>
  );
}

ImageAttachments.propTypes = {
  /* eslint-disable-next-line */
  attachments: PropTypes.array.isRequired,
};
