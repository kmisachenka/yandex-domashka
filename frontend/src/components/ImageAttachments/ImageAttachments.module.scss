@import '../../media';

.list {
  @include list-style-none;
  align-items: flex-end;

  display: flex;
  justify-content: flex-end;

  overflow: hidden; // TODO сделать прокрутку

  padding-top: 0;
  padding-left: 12px;
  padding-right: 12px;
  padding-bottom: 0;

  width: 100%;
}

.item {
  padding-right: 12px;

  &:last-child {
    padding-right: 0;
  }
}

.image {
  border: 1px solid rgba(218, 220, 224, 0.5);
  border-radius: 4px;
  max-height: 64px;

  max-width: 64px;
}

.svg {
  align-self: center;
  margin-left: 12px;
  margin-right: 12px;
}
