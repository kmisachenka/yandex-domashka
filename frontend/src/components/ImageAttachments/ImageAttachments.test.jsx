import React from 'react';
import { shallow } from 'enzyme';

import ImageAttachments from './ImageAttachments';

describe('ImageAttachments', () => {
  it('should match snapshot', () => {
    const attachments = [
      {
        type: 'image',
        url: 'https://link.com/image.png',
      },
    ];
    const wrapper = shallow(<ImageAttachments attachments={attachments} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a list', () => {
    const attachments = [
      {
        type: 'image',
        url: 'https://link.com/image.png',
      },
      {
        type: 'image',
        url: 'https://link.net/image.svg',
      },
    ];
    const wrapper = shallow(<ImageAttachments attachments={attachments} />);
    const list = wrapper.find('ul');
    const listItems = list.find('li');
    expect(listItems).toHaveLength(2);
  });
});
