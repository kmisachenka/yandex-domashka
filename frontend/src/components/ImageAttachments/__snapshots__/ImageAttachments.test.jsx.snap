// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageAttachments should match snapshot 1`] = `
<Fragment>
  <ForwardRef
    className="svg"
  />
  <ul
    className="list"
  >
    <li
      className="item"
      key="https://link.com/image.png"
    >
      <img
        alt="Картинка"
        className="image"
        src="/img/https://link.com/image.png"
      />
    </li>
  </ul>
</Fragment>
`;
